lake/commits/store.go

```diff
@@ -86,6 +87,12 @@ func (s *Store) Snapshot(ctx context.Context, leaf ksuid.KSUID) (*Snapshot, erro
 	if snap, ok := s.snapshots[leaf]; ok {
 		return snap, nil
 	}
+	if snap, err := s.getSnapshot(ctx, leaf); err != nil && !errors.Is(err, fs.ErrNotExist) {
+		return nil, err
+	} else if err == nil {
+		s.snapshots[leaf] = snap
+		return snap, nil
+	}
 	var objects []*Object
 	var base *Snapshot
 	for at := leaf; at != ksuid.Nil; {
```

Comment:
Should we be also doing `getSnapshots` when we're reversing through history?

Reply:
I'd been thinking we shouldn't because that would add extra latency, but it won't if we fetch the next object and next snapshot concurrently, so let me see how it looks.